### ISSUE TITLE
replay doesn't show cars 

### ISSUE THREAD

**Issue by traderforce** [cityflow-project/CityFlow#124]
![image](https://user-images.githubusercontent.com/24294874/115207562-4b5ebe80-a12e-11eb-8dfc-cc6d87bfbb18.png)
I run a example test program and  I want to see the simulation situation on the index.html. The signal changes can be seen but there are no cars on the road. Can anybody say what the possible  reasons are? Thanks!

**Comment by luxueliang1996** [comment]
I have the same problem

**Comment by 08doudou** [comment]
I have solved this problem. It means the replay.txt file which you generate by your own model lacks of two important values: the length and the width of this vehicle. You can compare the replay.txt you generate with the example files. Good luck!

**Comment by traderforce** [comment]
> I have solved this problem. It means the replay.txt file which you generate by your own model lacks of two important values: the length and the width of this vehicle. You can compare the replay.txt you generate with the example files. Good luck!

I also got this, thanks for your reply. It's really wierd.

**Comment by Fly-Pluche** [comment]
@traderforce @luxueliang1996 @traderforce Did you solved it?

**Comment by Fly-Pluche** [comment]
I didn't know the root cause, but I wrote a script to solve it
```
path = r'D:\workspace\CityFlow\frontend\1'
filename=r'\replayLogFile.txt'


with open(path+r'\2.txt','w')as final:
    with open(path+filename, 'r')as f:
        a=f.read().split(',')
        for i in a:
            if 'flow'in i:
                final.write(i+' 5.0 2.0,')
            else:
                final.write(i+',')
```

"length": 5.0,
"width": 2.0
It's defined in flow.json.

**Comment by MPHarryZhang** [comment]
> I have the same problem

Hi, I also had this problem, got roadnetLogFile and replayLogFile, and the road, the number of vehicles, the current step is displayed correctly, but, the same vehicles are not displayed in the replay, I checked the page, it is consistent with yours, in google There is only one warning in the browser: DevTools failed to load source map: Could not load content for file:///D:/Harry/WangZhiwen/CityFlow-master/CityFlow-master/frontend/pixi.min.js.map : System error: net::ERR_FILE_NOT_FOUND, before that, I have downloaded the js to the local and modified the script reference path in html, otherwise the browser will explode the error of net::ERR_FILE_NOT_FOUND, I hope to get your advice, maybe What are the reasons why the vehicle does not show up?

**Comment by MPHarryZhang** [comment]
> 我已经解决了这个问题。这意味着你自己的模型生成的replay.txt文件缺少两个重要的值：这辆车的长度和宽度。您可以将生成的 replay.txt 与示例文件进行比较。祝你好运！

Hello, can you tell me how to solve this problem?

**Comment by MPHarryZhang** [comment]
> > I have solved this problem. It means the replay.txt file which you generate by your own model lacks of two important values: the length and the width of this vehicle. You can compare the replay.txt you generate with the example files. Good luck!
> 
> I also got this, thanks for your reply. It's really wierd.

Hello, I also have this problem. Can you tell me how to solve it through codes?

**Comment by zhc134** [comment]
Did you specify length and width in `flow.json` file?

**Comment by MPHarryZhang** [comment]
> 您是否在文件中指定了长度和宽度`flow.json`？

Yes, the flow.json file has width and length. And I found that some algorithms run the results to display the vehicle normally, and some don't.

**Comment by zhc134** [comment]
> > 您是否在文件中指定了长度和宽度`flow.json`？
> 
> Yes, the flow.json file has width and length. And I found that some algorithms run the results to display the vehicle normally, and some don't.

Could you please give more detail about what kind of algorithm generates abnormal behavior?

**Comment by MPHarryZhang** [comment]
> > > 您是否在文件中指定了长度和内容`flow.json`？
> > 
> > 
> > 是的，flow.json 文件有宽度和长度。我发现有些算法运行结果可以正常显示车辆，有些则不能。
> 
> 您能否详细说明哪种算法会产生异常行为？

When I run CoLight, CoLight can't display the vehicle, but MPLight can do it normally. Now my research is mainly running these two algorithms.

**Comment by zhc134** [comment]
Weird, different algorithms should not influence replay behavior, there might be some problem in Python code or settings in config file, e.g. `saveReplay` should be true.

**Comment by MPHarryZhang** [comment]
> Weird, different algorithms should not influence replay behavior, there might be some problem in Python code or settings in config file, e.g. `saveReplay` should be true.

I'm also confused, the CoLight replay behavior is normal, it's just missing the display vehicle, but the length and width are valuable, which is confusing to me. I like this research direction very much, which is in progress as my master's research direction. I want a gym version of cityflow environment for multi-intersection traffic signal control. I don't know if there is a reusable work project, or I need to do it myself Work?

**Comment by zhc134** [comment]
checkout this repo `https://github.com/zhc134/tlc-baselines`

**Comment by MPHarryZhang** [comment]
> checkout this repo `https://github.com/zhc134/tlc-baselines`

Thank you very much.
